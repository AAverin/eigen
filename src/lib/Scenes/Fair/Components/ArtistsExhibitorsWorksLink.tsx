import { Box, color, Flex, Sans, Separator, space } from "@artsy/palette"
import React from "react"
import { TouchableOpacity } from "react-native"
import styled from "styled-components/native"

interface Props {
  onViewAllExhibitorsPressed: () => void
  onViewAllArtworksPressed: () => void
  onViewAllArtistsPressed: () => void
}

// ts-lint:ignore
const TabWrapper = styled(Flex)`
  margin-top: ${space(1)};
  margin-bottom: ${space(2)};
  flex-direction: row;
  justify-content: space-evenly;
`
const LeftTab = styled(Box)`
  border-right-width: 1px;
  padding-right: 25px;
  border-right-color: ${color("black10")};
`

export class ArtistsExhibitorsWorksLink extends React.Component<Props> {
  render() {
    const { onViewAllExhibitorsPressed, onViewAllArtistsPressed, onViewAllArtworksPressed } = this.props
    return (
      <>
        <TabWrapper>
          <TouchableOpacity onPress={() => onViewAllArtistsPressed()}>
            <LeftTab>
              <Sans size="3">Artists</Sans>
            </LeftTab>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => onViewAllExhibitorsPressed()}>
            <LeftTab right="3">
              <Sans size="3">Exhibitors</Sans>
            </LeftTab>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => onViewAllArtworksPressed()}>
            <Box right="3">
              <Sans size="3">Works</Sans>
            </Box>
          </TouchableOpacity>
        </TabWrapper>
        <Separator mt="1" />
      </>
    )
  }
}
