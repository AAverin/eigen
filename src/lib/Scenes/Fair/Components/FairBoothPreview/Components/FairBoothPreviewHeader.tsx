import { Box, Flex, Sans, Serif, space } from "@artsy/palette"
import InvertedButton from "lib/Components/Buttons/InvertedButton"
import OpaqueImageView from "lib/Components/OpaqueImageView"
import React from "react"
import { Dimensions, TouchableOpacity } from "react-native"
import styled from "styled-components/native"

interface Props {
  name: string
  location: string
  isFollowed?: boolean
  onFollowPartner?: () => void
  isFollowedChanging?: boolean
  url: string
  onViewFairBoothPressed: () => void
}

const roundedImageSize = 45

export const FairBoothPreviewHeader: React.SFC<Props> = ({
  name,
  location,
  isFollowed,
  onFollowPartner,
  isFollowedChanging,
  url,
  onViewFairBoothPressed,
}) => {
  const buttonWidth = 102
  const partnerNameWidth = Dimensions.get("window").width - buttonWidth - roundedImageSize - space(4) - space(1)
  return (
    <Flex justifyContent="space-between" alignItems="center" flexDirection="row" mb={1}>
      <TouchableOpacity onPress={() => onViewFairBoothPressed()}>
        <ImageAndTextWrapper flexDirection="row" alignItems="center">
          <Box mr={1}>{url && <RoundedImage imageURL={url} aspectRatio={1} />}</Box>
          <Box width={partnerNameWidth}>
            <TightendSerif size="2">{name}</TightendSerif>
            {location && (
              <TightendSans size="2" color="black60">
                {location}
              </TightendSans>
            )}
          </Box>
        </ImageAndTextWrapper>
      </TouchableOpacity>
      <Box width={102} height={34}>
        <InvertedButton
          grayBorder={true}
          text={isFollowed ? "Following" : "Follow"}
          onPress={onFollowPartner}
          selected={isFollowed}
          inProgress={isFollowedChanging}
        />
      </Box>
    </Flex>
  )
}

const RoundedImage = styled(OpaqueImageView)`
  height: ${roundedImageSize};
  width: ${roundedImageSize};
  border-radius: 22.5;
  overflow: hidden;
`

const TightendSerif = styled(Serif)`
  position: relative;
  top: 2;
`

const TightendSans = styled(Sans)`
  position: relative;
  top: -2;
`

const ImageAndTextWrapper = styled(Flex)`
  flex-wrap: nowrap;
`
