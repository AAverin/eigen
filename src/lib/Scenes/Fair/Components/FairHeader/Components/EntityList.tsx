import { Flex, Sans } from "@artsy/palette"
import React from "react"
import { TouchableOpacity } from "react-native"

interface Item {
  name: string
  href: string
}

interface EntityListProps {
  prefix: string
  list: Item[]
  displayedItems?: number
  count: number
}

interface EntityItemProps {
  item: Item
  isFirst: boolean
  isLast: boolean
}

export const EntityItem: React.SFC<EntityItemProps> = ({ item: { name, href }, isFirst, isLast }) => {
  let text = `${name}, `

  if ((isFirst && isLast) || isLast) {
    text = name
  }

  return (
    <TouchableOpacity onPress={() => this.handlePress(this, href)}>
      <Sans weight="medium" size="3" lineHeight="19">
        {text}
      </Sans>
    </TouchableOpacity>
  )
}

export const EntityList: React.SFC<EntityListProps> = ({ prefix, list, displayedItems = 1, count }) => {
  if (!list.length) {
    return null
  }

  const filteredList = list.slice(0, displayedItems)
  const shouldShowMore = count > displayedItems

  return (
    <Flex flexDirection="row" flexWrap="wrap" mb="8">
      <Sans size="3" lineHeight="19">
        {prefix + " "}
      </Sans>
      {filteredList.map((item, i) => (
        <EntityItem item={item} isFirst={i === 0} isLast={i === filteredList.length - 1} />
      ))}
      {shouldShowMore && (
        <>
          <Sans size="3" lineHeight="19">
            {" and "}
          </Sans>
          <TouchableOpacity onPress={() => this.props.viewAllArtists()}>
            <Sans weight="medium" size="3" lineHeight="19">
              {count - displayedItems + " others"}
            </Sans>
          </TouchableOpacity>
        </>
      )}
    </Flex>
  )
}
