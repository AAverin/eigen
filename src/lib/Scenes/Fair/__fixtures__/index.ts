export const FairFixture = {
  id: "sofa-chicago-2018",
  name: "SOFA CHICAGO 2018",
  image: {
    image_url: "https://d32dm0rphc51dk.cloudfront.net/o0Zrtm-CgwbV1syD_33DuA/:version.jpg",
    aspect_ratio: 1,
    url: "https://d32dm0rphc51dk.cloudfront.net/o0Zrtm-CgwbV1syD_33DuA/square.jpg",
  },
  organizer: {
    profile: {
      icon: {
        id: "58ff95298b3b8104783c119c",
        href: null,
        height: 279,
        width: 279,
        url: "https://d32dm0rphc51dk.cloudfront.net/B1x-CqoWO1KitQnRj7PrrA/untouched-png.png",
      },
      name: "SOFA CHICAGO",
      __id: "UHJvZmlsZTpzb2ZhLWNoaWNhZ28=",
    },
  },
  start_at: "2018-11-01T17:00:00+00:00",
  end_at: "2018-11-04T18:00:00+00:00",
  location: {
    address: "",
    address_2: "",
    coordinates: {
      lat: 41.891551,
      lng: -87.6073751,
    },
    display: "41.891972, -87.605161",
    city: "Chicago",
    state: "Illinois",
    country: "United States",
    postal_code: "60611",
  },
  __id: "RmFpcjpzb2ZhLWNoaWNhZ28tMjAxOA==",
}
