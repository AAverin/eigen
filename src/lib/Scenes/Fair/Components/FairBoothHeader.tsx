import { Box, Sans, Serif, Spacer } from "@artsy/palette"
import { FairBoothHeader_show } from "__generated__/FairBoothHeader_show.graphql"
import { InvertedButton } from "lib/Components/Buttons"
import React from "react"
import { TouchableOpacity } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"

interface Props {
  show: FairBoothHeader_show
  onSaveShowPressed: () => void
  onTitlePressed: (id: string) => void
}

const formatCounts = ({ artists, artworks }) => {
  const artistLabel = artists === 1 ? "artist" : "artists"
  const worksLabel = artworks === 1 ? "work" : "works"
  return `${artworks} ${worksLabel} by ${artists} ${artistLabel}`
}

export const FairBoothHeader: React.SFC<Props> = ({
  show: {
    partner: { name: partnerName, id: partnerId },
    fair: { name: fairName },
    counts,
  },
  onSaveShowPressed,
  onTitlePressed,
}) => {
  return (
    <Box pt={12} px={2}>
      <TouchableOpacity onPress={() => onTitlePressed(partnerId)}>
        <Serif size="8">{partnerName}</Serif>
      </TouchableOpacity>
      <Spacer m={0.3} />
      <Sans weight="medium" size="3t">
        {fairName} â€¢ Booth
      </Sans>
      <Spacer m={0.5} />
      <Serif size="3t">Galleries</Serif>
      <Spacer m={0.3} />
      <Sans size="3t" color="black60">
        {formatCounts(counts)}
      </Sans>
      <Spacer m={6} />
      <InvertedButton text="Follow gallery" onPress={() => onSaveShowPressed()} />
    </Box>
  )
}

export const FairBoothHeaderContainer = createFragmentContainer(
  FairBoothHeader,
  graphql`
    fragment FairBoothHeader_show on Show {
      fair {
        name
      }
      partner {
        ... on Partner {
          name
          id
        }
        ... on ExternalPartner {
          name
          id
        }
      }
      counts {
        artworks
        artists
      }
    }
  `
)
