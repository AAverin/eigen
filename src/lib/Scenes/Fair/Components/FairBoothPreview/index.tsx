import { Box } from "@artsy/palette"
import { FairBoothPreview_show } from "__generated__/FairBoothPreview_show.graphql"
import GenericGrid from "lib/Components/ArtworkGrids/GenericGrid"
import { CaretButton } from "lib/Components/Buttons/CaretButton"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { FairBoothPreviewHeader } from "./Components/FairBoothPreviewHeader"

interface Props {
  show: FairBoothPreview_show
  onViewFairBoothPressed: () => void
  Component: any
}

export const FairBoothPreviewContainer = createFragmentContainer<Props>(
  props => {
    const {
      show: { artworks_connection, cover_image, location, partner },
      onViewFairBoothPressed,
      Component,
    } = props
    const display = !!location ? location.display : ""

    return (
      <Box my={1}>
        <FairBoothPreviewHeader
          context={Component}
          name={partner.name}
          location={display}
          url={cover_image && cover_image.url}
          galleryURL={partner.href}
        />
        <Box mt={1}>{<GenericGrid artworks={artworks_connection.edges.map(a => a.node) as any} />}</Box>
        <Box mt={2}>
          <CaretButton text="View all works" onPress={() => onViewFairBoothPressed()} />
        </Box>
      </Box>
    )
  },
  graphql`
    fragment FairBoothPreview_show on Show {
      id
      name
      is_fair_booth

      ...FairBooth_show

      partner {
        ... on Partner {
          name
          href
        }

        ... on ExternalPartner {
          name
        }
      }

      fair {
        name
      }

      cover_image {
        url
      }

      location {
        display
      }

      artworks_connection(first: 4) {
        edges {
          node {
            ...GenericGrid_artworks
          }
        }
      }
    }
  `
)
