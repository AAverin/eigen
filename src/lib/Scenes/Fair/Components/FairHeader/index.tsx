import { Box, Flex, Sans, space, Spacer } from "@artsy/palette"
import { FairHeader_fair } from "__generated__/FairHeader_fair.graphql"
import { FairHeaderMutation } from "__generated__/FairHeaderMutation.graphql"
import InvertedButton from "lib/Components/Buttons/InvertedButton"
import { EntityList } from "lib/Components/EntityList"
import OpaqueImageView from "lib/Components/OpaqueImageView"
import Switchboard from "lib/NativeModules/SwitchBoard"
import moment from "moment"
import React from "react"
import { Dimensions, Image } from "react-native"
import { commitMutation, createFragmentContainer, graphql, RelayProp } from "react-relay"
import styled from "styled-components/native"
import { CountdownTimer } from "./CountdownTimer"

interface Props {
  fair: FairHeader_fair
  onSaveShowPressed?: () => Promise<void>
  viewAllExhibitors: () => void
  viewAllArtists: () => void
  relay: RelayProp
}

interface State {
  isSavedFairStateUpdating: boolean
}

const BackgroundImage = styled(OpaqueImageView)<{ height: number; width: number }>`
  position: absolute;
  height: 100%;
  width: 100%;
`

// Set background color of overlay based on logo color
const Overlay = styled.View`
  background-color: rgba(0, 0, 0, 0.3);
  width: 100%;
  height: 100%;
  position: absolute;
`

const Logo = styled(Image)`
  width: 150;
  height: 150;
  margin-left: auto;
  margin-right: auto;
  background-color: transparent;
  margin-bottom: ${space(1)};
  tint-color: white;
`

const CountdownContainer = styled.View`
  position: absolute;
  bottom: ${space(6)};
  left: 0;
  width: 100%;
`

export class FairHeader extends React.Component<Props, State> {
  state = { isSavedFairStateUpdating: false }

  getContextualDetails() {
    const { viewAllArtists, viewAllExhibitors } = this.props
    const { artists_names, counts, partner_names } = this.props.fair

    const artistList = artists_names.edges.map(i => i.node).filter(Boolean)
    const partnerList = partner_names.edges
      .map(i => {
        if (i.node.partner && i.node.partner.profile && i.node.partner.profile.name) {
          return {
            href: "show/" + i.node.id,
            name: i.node.partner.profile.name,
          }
        }
      })
      .filter(Boolean)

    return (
      <>
        <EntityList
          prefix="Works by"
          list={artistList}
          count={counts.artists}
          displayedItems={2}
          onItemSelected={this.handlePress}
          onViewAllPressed={viewAllArtists}
        />
        <EntityList
          prefix="From"
          list={partnerList}
          count={counts.partners}
          displayedItems={2}
          onItemSelected={this.handlePress}
          onViewAllPressed={viewAllExhibitors}
        />
      </>
    )
  }

  handlePress = item => {
    Switchboard.presentNavigationViewController(this, item)
  }

  handleSaveFair() {
    const {
      relay,
      fair: {
        profile: { __id: fairProfileID, id: fairID, is_followed: isFairFollowed },
      },
    } = this.props

    this.setState(
      {
        isSavedFairStateUpdating: true,
      },
      () => {
        if (fairProfileID) {
          return commitMutation<FairHeaderMutation>(relay.environment, {
            onCompleted: () => {
              this.setState({
                isSavedFairStateUpdating: false,
              })
            },
            mutation: graphql`
              mutation FairHeaderMutation($input: FollowProfileInput!) {
                followProfile(input: $input) {
                  profile {
                    id
                    is_followed
                    __id
                  }
                }
              }
            `,
            variables: {
              input: {
                profile_id: fairProfileID,
                unfollow: isFairFollowed,
              },
            },
            optimisticResponse: {
              followProfile: {
                profile: {
                  __id: fairProfileID,
                  is_followed: !isFairFollowed,
                  id: fairID,
                },
              },
            },
            updater: store => {
              store.get(fairProfileID).setValue(!isFairFollowed, "is_followed")
            },
          })
        }
      }
    )
  }

  render() {
    const {
      fair: { image, name, profile, start_at, end_at },
    } = this.props
    const { width: screenWidth } = Dimensions.get("window")
    const imageHeight = 567
    const { isSavedFairStateUpdating } = this.state

    return (
      <>
        <Box style={{ height: imageHeight, width: screenWidth, position: "relative" }}>
          <BackgroundImage imageURL={image.url} height={imageHeight} width={screenWidth} />
          <Overlay />
          <Flex flexDirection="row" justifyContent="center" alignItems="center" px={2} height={imageHeight}>
            <Flex alignItems="center" flexDirection="column" flexGrow={1}>
              {profile && <Logo source={{ uri: profile.icon.url }} />}
              <Sans size="3t" weight="medium" textAlign="center" color="white100">
                {name}
              </Sans>
              <Sans size="3" textAlign="center" color="white100">
                {moment(start_at).format("MMM Do")} - {moment(end_at).format("MMM Do")}
              </Sans>
            </Flex>
          </Flex>
        </Box>
        <CountdownContainer>
          <CountdownTimer startAt={start_at} endAt={end_at} />
        </CountdownContainer>
        <Spacer mt={2} />
        <Box mx={2}>{this.getContextualDetails()}</Box>
        {profile && (
          <Box px={2}>
            <Spacer m={2} mt={1} />
            <InvertedButton
              text={profile.is_followed ? "Fair saved" : "Save fair"}
              onPress={() => this.handleSaveFair()}
              selected={profile.is_followed}
              inProgress={isSavedFairStateUpdating}
              grayBorder={true}
              buttonSize={"large"}
            />
            <Spacer m={1} />
          </Box>
        )}
      </>
    )
  }
}

export const FairHeaderContainer = createFragmentContainer(
  FairHeader,
  graphql`
    fragment FairHeader_fair on Fair {
      id
      name
      counts {
        artists
        partners
      }
      partner_names: shows_connection(first: 2) {
        edges {
          node {
            id
            partner {
              ... on Partner {
                profile {
                  name
                }
              }
            }
          }
        }
      }
      artists_names: artists(first: 2) {
        edges {
          node {
            name
            href
          }
        }
      }

      image {
        image_url
        aspect_ratio
        url
      }

      profile {
        icon {
          id
          href
          height
          width
          url
        }
        __id
        id
        name
        is_followed
      }

      start_at
      end_at
    }
  `
)
