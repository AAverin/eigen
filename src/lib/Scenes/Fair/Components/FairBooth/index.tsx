import { Box } from "@artsy/palette"
import { FairBooth_show } from "__generated__/FairBooth_show.graphql"
import GenericGrid from "lib/Components/ArtworkGrids/GenericGrid"
import { CaretButton } from "lib/Components/Buttons/CaretButton"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { FairBoothHeader } from "./Components/FairBoothHeader"

interface Props {
  show: FairBooth_show
}

export const FairBoothContainer = createFragmentContainer<Props>(
  props => {
    const {
      show: { artworks_connection, cover_image, location, partner },
    } = props

    const display = !!location ? location.display : ""
    return (
      <Box my={1}>
        <FairBoothHeader name={partner.name} location={display} url={cover_image && cover_image.url} />
        <Box mt={1}>{<GenericGrid artworks={artworks_connection.edges.map(a => a.node) as any} />}</Box>
        <Box mt={2}>
          <CaretButton text="View all works" />
        </Box>
      </Box>
    )
  },
  graphql`
    fragment FairBooth_show on Show {
      id
      name
      is_fair_booth

      partner {
        ... on Partner {
          name
        }

        ... on ExternalPartner {
          name
        }
      }

      fair {
        name
      }

      cover_image {
        url
      }

      location {
        display
      }

      artworks_connection(first: 4) {
        edges {
          node {
            ...GenericGrid_artworks
          }
        }
      }
    }
  `
)
