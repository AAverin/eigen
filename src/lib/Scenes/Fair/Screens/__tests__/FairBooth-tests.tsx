import { FairBoothShowFixture } from "lib/__fixtures__/FairBoothShowFixture"
import { MockRelayRenderer } from "lib/tests/MockRelayRenderer"
import { renderUntil } from "lib/tests/renderUntil"
import React from "react"
import { graphql } from "react-relay"
import { FairBoothHeaderContainer as FairBoothHeader } from "../../Components/FairBoothHeader"
import { FairBoothContainer as FairBooth } from "../FairBooth"

jest.unmock("react-relay")

// This test is intentionally disabled as of RN 0.59 upgrade
xit("renders properly", async () => {
  const tree = await renderUntil(
    wrapper => wrapper.find(FairBoothHeader).length > 0,
    <MockRelayRenderer
      Component={({ show }) => <FairBooth show={show} />}
      query={graphql`
        query FairBoothTestsQuery {
          show(id: "two-palms-two-palms-at-art-basel-miami-beach-2018") {
            ...FairBooth_show
          }
        }
      `}
      mockResolvers={{
        Show: () => FairBoothShowFixture,
      }}
    />
  )
  expect(tree.html()).toMatchSnapshot()
})
