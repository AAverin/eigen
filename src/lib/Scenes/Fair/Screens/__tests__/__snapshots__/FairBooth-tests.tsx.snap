// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `"<rctscrollview data=\\"[object Object],[object Object]\\" listheadercomponent=\\"[object Object]\\" initialnumtorender=\\"10\\" maxtorenderperbatch=\\"10\\" scrolleventthrottle=\\"50\\" updatecellsbatchingperiod=\\"50\\" windowsize=\\"21\\" numcolumns=\\"1\\" viewabilityconfigcallbackpairs=\\"\\" stickyheaderindices=\\"\\"><view><view><view pt=\\"12\\" px=\\"2\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"32px\\" lineheight=\\"38px\\" class=\\"sc-bwzfXH uTxCW\\" ellipsizemode=\\"tail\\">Two Palms</text><view m=\\"0.3\\"></view><text font-family=\\"Unica77LL-Medium\\" font-size=\\"14px\\" lineheight=\\"20px\\" class=\\"sc-bdVaJa bDWFJH\\" ellipsizemode=\\"tail\\">Art Basel in Miami Beach 2018 â€¢ Booth</text><view m=\\"0.5\\"></view><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"20px\\" class=\\"sc-bwzfXH uTxCW\\" ellipsizemode=\\"tail\\">Galleries</text><view m=\\"0.3\\"></view><text font-family=\\"Unica77LL-Regular\\" font-size=\\"14px\\" lineheight=\\"20px\\" class=\\"sc-bdVaJa bDWFJH\\" color=\\"black60\\" ellipsizemode=\\"tail\\">26 works by 7 artists</text><view m=\\"6\\"></view><view style=\\"justify-content: center; align-items: center; border-width: 1px; border-radius: 2px; height: 46px; background-color: rgb(0, 0, 0); opacity: 1;\\"><text font-family=\\"Unica77LL-Medium\\" font-size=\\"14px\\" lineheight=\\"24px\\" class=\\"sc-bdVaJa bDWFJH\\" color=\\"white\\" ellipsizemode=\\"tail\\">Follow gallery</text></view></view></view><view><view px=\\"2\\" py=\\"2\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"26px\\" lineheight=\\"32px\\" class=\\"sc-bwzfXH uTxCW\\" mt=\\"2\\" mb=\\"3\\" ellipsizemode=\\"tail\\">All works</text><view><view style=\\"flex-direction: row;\\" accessibilitylabel=\\"Artworks Content View\\"></view></view><view style=\\"opacity: 1;\\"><text font-family=\\"Unica77LL-Medium\\" font-size=\\"14px\\" lineheight=\\"24px\\" class=\\"sc-bdVaJa bDWFJH\\" my=\\"2\\" ellipsizemode=\\"tail\\">View all works</text></view></view></view><view><view px=\\"2\\" py=\\"2\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"22px\\" lineheight=\\"32px\\" class=\\"sc-bwzfXH uTxCW\\" ellipsizemode=\\"tail\\">Artists</text><view m=\\"1\\"></view><view style=\\"opacity: 1;\\"><view justifycontent=\\"space-between\\" alignitems=\\"center\\" flexdirection=\\"row\\"><aropaqueimageview imageurl=\\"https://d32dm0rphc51dk.cloudfront.net/U0Ap3ZESNaZ1XHJQr8pmhw/tall.jpg\\" aspectratio=\\"1\\"></aropaqueimageview><view m=\\"1\\"></view><view flexdirection=\\"column\\" alignitems=\\"flex-start\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"20px\\" class=\\"sc-bwzfXH uTxCW\\" ellipsizemode=\\"tail\\">Cecily Brown</text><text font-family=\\"Unica77LL-Regular\\" font-size=\\"14px\\" lineheight=\\"20px\\" class=\\"sc-bdVaJa bDWFJH\\" color=\\"black60\\" ellipsizemode=\\"tail\\">British, b. 1969</text></view><view m=\\"1\\"></view><view width=\\"102\\" height=\\"34\\"><view style=\\"justify-content: center; align-items: center; border-radius: 2px; background-color: rgb(0, 0, 0); border-width: 1px;\\"><view><text ellipsizemode=\\"tail\\">Follow</text></view></view></view></view></view><view m=\\"1\\"></view><view style=\\"opacity: 1;\\"><view justifycontent=\\"space-between\\" alignitems=\\"center\\" flexdirection=\\"row\\"><aropaqueimageview imageurl=\\"https://d32dm0rphc51dk.cloudfront.net/J98XhW9fvRVV-IDdAFY_ig/tall.jpg\\" aspectratio=\\"1\\"></aropaqueimageview><view m=\\"1\\"></view><view flexdirection=\\"column\\" alignitems=\\"flex-start\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"20px\\" class=\\"sc-bwzfXH uTxCW\\" ellipsizemode=\\"tail\\">Jeff Koons</text><text font-family=\\"Unica77LL-Regular\\" font-size=\\"14px\\" lineheight=\\"20px\\" class=\\"sc-bdVaJa bDWFJH\\" color=\\"black60\\" ellipsizemode=\\"tail\\">American, b. 1955</text></view><view m=\\"1\\"></view><view width=\\"102\\" height=\\"34\\"><view style=\\"justify-content: center; align-items: center; border-radius: 2px; background-color: rgb(0, 0, 0); border-width: 1px;\\"><view><text ellipsizemode=\\"tail\\">Follow</text></view></view></view></view></view><view m=\\"1\\"></view><view style=\\"opacity: 1;\\"><view justifycontent=\\"space-between\\" alignitems=\\"center\\" flexdirection=\\"row\\"><aropaqueimageview imageurl=\\"https://d32dm0rphc51dk.cloudfront.net/gpxuOQYgaDxHycQRA9ZBSQ/tall.jpg\\" aspectratio=\\"1\\"></aropaqueimageview><view m=\\"1\\"></view><view flexdirection=\\"column\\" alignitems=\\"flex-start\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"20px\\" class=\\"sc-bwzfXH uTxCW\\" ellipsizemode=\\"tail\\">Elizabeth Peyton</text><text font-family=\\"Unica77LL-Regular\\" font-size=\\"14px\\" lineheight=\\"20px\\" class=\\"sc-bdVaJa bDWFJH\\" color=\\"black60\\" ellipsizemode=\\"tail\\">American, b. 1965</text></view><view m=\\"1\\"></view><view width=\\"102\\" height=\\"34\\"><view style=\\"justify-content: center; align-items: center; border-radius: 2px; background-color: rgb(0, 0, 0); border-width: 1px;\\"><view><text ellipsizemode=\\"tail\\">Follow</text></view></view></view></view></view><view m=\\"1\\"></view><view style=\\"opacity: 1;\\"><view justifycontent=\\"space-between\\" alignitems=\\"center\\" flexdirection=\\"row\\"><aropaqueimageview imageurl=\\"https://d32dm0rphc51dk.cloudfront.net/SjhBVNSH4IzXyPJj7sa2YA/tall.jpg\\" aspectratio=\\"1\\"></aropaqueimageview><view m=\\"1\\"></view><view flexdirection=\\"column\\" alignitems=\\"flex-start\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"20px\\" class=\\"sc-bwzfXH uTxCW\\" ellipsizemode=\\"tail\\">Stanley Whitney</text><text font-family=\\"Unica77LL-Regular\\" font-size=\\"14px\\" lineheight=\\"20px\\" class=\\"sc-bdVaJa bDWFJH\\" color=\\"black60\\" ellipsizemode=\\"tail\\">American, b. 1946</text></view><view m=\\"1\\"></view><view width=\\"102\\" height=\\"34\\"><view style=\\"justify-content: center; align-items: center; border-radius: 2px; background-color: rgb(0, 0, 0); border-width: 1px;\\"><view><text ellipsizemode=\\"tail\\">Follow</text></view></view></view></view></view><view m=\\"1\\"></view><view style=\\"opacity: 1;\\"><view justifycontent=\\"space-between\\" alignitems=\\"center\\" flexdirection=\\"row\\"><aropaqueimageview imageurl=\\"https://d32dm0rphc51dk.cloudfront.net/6xsCdFDWx3eFePd1LrzX0Q/tall.jpg\\" aspectratio=\\"1\\"></aropaqueimageview><view m=\\"1\\"></view><view flexdirection=\\"column\\" alignitems=\\"flex-start\\"><text font-family=\\"AGaramondPro-Regular\\" font-size=\\"16px\\" lineheight=\\"20px\\" class=\\"sc-bwzfXH uTxCW\\" ellipsizemode=\\"tail\\">Mel Bochner</text><text font-family=\\"Unica77LL-Regular\\" font-size=\\"14px\\" lineheight=\\"20px\\" class=\\"sc-bdVaJa bDWFJH\\" color=\\"black60\\" ellipsizemode=\\"tail\\">American, b. 1940</text></view><view m=\\"1\\"></view><view width=\\"102\\" height=\\"34\\"><view style=\\"justify-content: center; align-items: center; border-radius: 2px; background-color: rgb(0, 0, 0); border-width: 1px;\\"><view><text ellipsizemode=\\"tail\\">Follow</text></view></view></view></view></view><view m=\\"1\\"></view><view style=\\"opacity: 1;\\"><text font-family=\\"Unica77LL-Medium\\" font-size=\\"14px\\" lineheight=\\"24px\\" class=\\"sc-bdVaJa bDWFJH\\" my=\\"2\\" ellipsizemode=\\"tail\\">View all artists</text></view></view></view><view style=\\"height: 0px;\\"></view></view></rctscrollview>"`;
